handlers/validator.go

```diff
@@ -796,34 +799,25 @@ func Validator(w http.ResponseWriter, r *http.Request) {
 			lastExportedEpoch := (lastStatsDay+1)*utils.EpochsPerDay() - 1
 			if actualSyncPeriods[0].LastEpoch > lastExportedEpoch {
 				lookback := int64(latestEpoch - lastExportedEpoch)
-				res, err := db.BigtableClient.GetValidatorSyncDutiesHistory([]uint64{index}, latestEpoch-uint64(lookback), latestEpoch)
+				syncStatsBt, err := db.BigtableClient.GetValidatorSyncCommitteesStats([]uint64{index}, latestEpoch-uint64(lookback), latestEpoch)
 				if err != nil {
 					return fmt.Errorf("error retrieving validator sync participations data from bigtable: %v", err)
 				}
-				for _, r := range res[index] {
-					slotTime := utils.SlotToTime(r.Slot)
-					if r.Status == 0 && time.Since(slotTime) > time.Minute {
-						r.Status = 2
-					}
-					switch r.Status {
-					case 0:
-						syncStats.ScheduledSync++
-					case 1:
-						syncStats.ParticipatedSync++
-					case 2:
-						syncStats.MissedSync++
-					case 3:
-						syncStats.OrphanedSync++
-					}
-				}
+				syncStats.MissedSync += syncStatsBt.MissedSlots
+				syncStats.ParticipatedSync += syncStatsBt.ParticipatedSlots
+				syncStats.ScheduledSync += syncStatsBt.ScheduledSlots
 			}
-			validatorPageData.ParticipatedSyncCount = syncStats.ParticipatedSync
-			validatorPageData.MissedSyncCount = syncStats.MissedSync
-			validatorPageData.OrphanedSyncCount = syncStats.OrphanedSync
-			validatorPageData.ScheduledSyncCount = syncStats.ScheduledSync
+
+			validatorPageData.SlotsDoneInCurrentSyncCommittee = validatorPageData.SlotsPerSyncCommittee - syncStats.ScheduledSync
+
+			validatorPageData.ParticipatedSyncCountSlots = syncStats.ParticipatedSync
+			validatorPageData.MissedSyncCountSlots = syncStats.MissedSync
+			validatorPageData.OrphanedSyncCountSlots = syncStats.OrphanedSync
+			validatorPageData.ScheduledSyncCountSlots = syncStats.ScheduledSync
 			// actual sync duty count and percentage
-			validatorPageData.SyncCount = validatorPageData.ParticipatedSyncCount + validatorPageData.MissedSyncCount + validatorPageData.OrphanedSyncCount + syncStats.ScheduledSync
-			validatorPageData.UnmissedSyncPercentage = float64(validatorPageData.SyncCount-validatorPageData.MissedSyncCount) / float64(validatorPageData.SyncCount)
+			validatorPageData.SyncCountSlots = validatorPageData.ParticipatedSyncCountSlots + validatorPageData.MissedSyncCountSlots + validatorPageData.OrphanedSyncCountSlots + validatorPageData.ScheduledSyncCountSlots
```

Comment:
SyncCountSlots isn't used anywhere else except the line below it so there is no need to make it a struct entry.

Reply:
Right, good call. Must have changed during implementation.

Reply:
It was deleted here but not in the struct **ValidatorPageData** itself.